tests/golos.publication_rewards_tests.cpp

```diff
@@ -378,21 +397,47 @@ class reward_calcs_tester : public golos_tester {
                         }
                     }
                     for (auto& r : cur_rewards) {
-                        _state.balances[r.first].vestamount += get_converted_to_vesting(r.second);
+                        if (_state.balances[r.first].vestclosed) {
+                            _state.balances[_forum_name].paymentsamount += (r.second + close_acc_delta);
+                            _state.balances[_forum_name].tokenamount += (r.second + close_acc_delta);
+                        }
+                        else {
+                            _state.balances[r.first].vestamount += get_converted_to_vesting(r.second);
+                        }
                         unclaimed_funds -= r.second;
                     }
 
                     auto author_payout =  payout - curation_payout;
                     double ben_payout_sum = 0.0;
                     for (auto& ben : m.beneficiaries) {
                         double ben_payout = author_payout * get_prop(ben.weight);
-                        _state.balances[ben.account].vestamount += get_converted_to_vesting(ben_payout);
+                        if (_state.balances[ben.account].vestclosed) {
+                            _state.balances[_forum_name].paymentsamount += (ben_payout + close_acc_delta);
+                            _state.balances[_forum_name].tokenamount += (ben_payout + close_acc_delta);
+                        }
+                        else {
+                            _state.balances[ben.account].vestamount += get_converted_to_vesting(ben_payout);
+                        }
                         ben_payout_sum += ben_payout;
                     }
                     author_payout -= ben_payout_sum;
 
-                    _state.balances[m.key.author].tokenamount += author_payout * m.tokenprop;
-                    _state.balances[m.key.author].vestamount += get_converted_to_vesting(author_payout * (1.0 - m.tokenprop));
+                    auto author_token_rwrd = author_payout * m.tokenprop;
+                    if (_state.balances[m.key.author].tokenclosed) {
+                        _state.balances[_forum_name].paymentsamount += author_token_rwrd;
+                        _state.balances[_forum_name].tokenamount += author_token_rwrd;
```

Comment:
It's very strange that you need added `author_token_rwrd` into `paymentsamount` and `tokenamount`.